File: frontend/src/objects/merchant/Merchant.ts
import type { MerchantCategory, ProductId, MerchantId } from '@/objects/shared/ids'
import type { Promotion } from '@/objects/shared/Promotion'
import type { MerchantBusinessStatus, MerchantHolidayBusinessHour, MerchantWeeklyBusinessHour } from './MerchantBusinessHours'

export interface Merchant {
  id: MerchantId
  storeName: string
  category: MerchantCategory
  address: string
  phone: string
  rating: number
  tags: string[]
  featuredProductIds: ProductId[]
  /** 店铺头图，http(s) 链接；未设置时后端可能省略或为 null */
  imageUrl?: string | null
  description: string
  announcement: string
  promotions?: Promotion[]
  businessStatus?: MerchantBusinessStatus
  weeklyBusinessHours?: MerchantWeeklyBusinessHour[]
  holidayBusinessHours?: MerchantHolidayBusinessHour[]
}

File: backend/src/merchant/objects/Merchant.scala
package delivery.merchant.objects

import delivery.shared.objects.{MerchantCategory, MerchantId, ProductId, Promotion}

final case class Merchant(
    id: MerchantId,
    storeName: String,
    category: MerchantCategory,
    address: String,
    phone: String,
    rating: Double,
    tags: List[String],
    featuredProductIds: List[ProductId],
    imageUrl: Option[String],
    description: String = "",
    announcement: String = "",
    promotions: List[Promotion] = Nil,
    businessStatus: String = "open",
    weeklyBusinessHours: List[MerchantWeeklyBusinessHour] = Nil,
    holidayBusinessHours: List[MerchantHolidayBusinessHour] = Nil
)
